api/lib/opentelemetry/trace/samplers/basic_decision.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Trace
+    module Samplers
+      # This class represents a BasicDecision, which is a utility class used
+      # by the {AlwaysSampleSampler} and the {NeverSampleSampler}. A
+      # BasicDecision can have a boolean sampling decision and will always
+      # return an empty set of attributes.
+      class BasicDecision
+        # Returns a new decision with the specified sampling decision and
+        # and empty attributes
+        #
+        # @param [Boolean] decision Whether a span should be sampled
+        def initialize(decision:)
+          @decision = decision
+        end
+
+        # Returns a sampling decision of whether this span should be sampled
+        #
+        # @return [Boolean] sampling decision
+        def sampled?
+          @decision
+        end
+
+        # Always returns an empty set of attributes
+        #
+        # @return [Hash<String, Object>]
+        def attributes
+          {}
```

Comment:
I *think* the attributes hash should be frozen. The spec isn't clear on this, but it implies the attributes are captured by the Decision rather than being modified afterwards.

Reply:
That was my instinct, but I couldn't find anything official or unofficial pointing towards an immutable data structure.

The obligatory cross reference to java shows `Map<String, AttributeValue>` return type: https://github.com/open-telemetry/opentelemetry-java/blob/6d1b31bfa639af653649d694ea5660fb7c3fd5ce/api/src/main/java/io/opentelemetry/trace/Sampler.java#L79-L86

Reply:
Reading deeper into the Javadocs, `Collections.emptyMap` is immutable. So, it might just be a matter of clarifying some of these details on the spec.

Reply:
Assuming that `Decision#attributes` can be immutable, and there are hints from the Java implementation that is possible, this is done.